src/java/org/apache/cassandra/db/CounterMutationCallback.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.net.Message;
+import org.apache.cassandra.net.MessagingService;
+import org.apache.cassandra.net.NoPayload;
+import org.apache.cassandra.net.SensorsCustomParams;
+import org.apache.cassandra.sensors.RequestSensors;
+import org.apache.cassandra.sensors.Sensor;
+import org.apache.cassandra.sensors.SensorsRegistry;
+import org.apache.cassandra.sensors.Type;
+
+/**
+ * A counter mutation callback that encapsulates {@link RequestSensors} and replica count
+ */
+public class CounterMutationCallback implements Runnable
+{
+    private final Message<CounterMutation> respondTo;
+    private final InetAddressAndPort respondToAddress;
+    private final RequestSensors requestSensors;
+    private int replicaCount = 0; // Default to CL.ANY
```

Comment:
I don't quite get this: the replica count should be independent from the CL, that is, it should always be equal to the number of replica being alive, while the CL only determines the number of replica we _wait_ for. I.e., for RF=3, this would be 3 most of the time, and regardless of the CL. What am I missing?

Reply:
Yes the comments are confusing. Will fix them.

FTR (and tangential to your comment) CL.ANY is not supported with counter mutations.